<a href="https://colab.research.google.com/github/arindamdeyofficial/houseprediction/blob/main/houseprediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
df = pd.read_csv('https://raw.githubusercontent.com/arindamdeyofficial/houseprediction/main/housing.csv')


In [ ]:
df.shape
df.head()

longitude  latitude  ...  median_house_value  ocean_proximity
0    -122.23     37.88  ...            452600.0         NEAR BAY
1    -122.22     37.86  ...            358500.0         NEAR BAY
2    -122.24     37.85  ...            352100.0         NEAR BAY
3    -122.25     37.85  ...            341300.0         NEAR BAY
4    -122.25     37.85  ...            342200.0         NEAR BAY

[5 rows x 10 columns]

In [ ]:
df.columns

Index(['longitude', 'latitude', 'housing_median_age', 'total_rooms',
       'total_bedrooms', 'population', 'households', 'median_income',
       'median_house_value', 'ocean_proximity'],
      dtype='object')

In [ ]:
df.groupby('ocean_proximity').count()

longitude  latitude  ...  median_income  median_house_value
ocean_proximity                       ...                                   
<1H OCEAN             9136      9136  ...           9136                9136
INLAND                6551      6551  ...           6551                6551
ISLAND                   5         5  ...              5                   5
NEAR BAY              2290      2290  ...           2290                2290
NEAR OCEAN            2658      2658  ...           2658                2658

[5 rows x 9 columns]

In [58]:
df.isna().any()

longitude             False
latitude              False
housing_median_age    False
total_rooms           False
total_bedrooms         True
population            False
households            False
median_income         False
median_house_value    False
ocean_proximity       False
dtype: bool

In [ ]:
df.count()

longitude             20640
latitude              20640
housing_median_age    20640
total_rooms           20640
total_bedrooms        20433
population            20640
households            20640
median_income         20640
median_house_value    20640
ocean_proximity       20640
dtype: int64

In [86]:
df[df.columns[df.isna().any()]]

total_bedrooms
0               129.0
1              1106.0
2               190.0
3               235.0
4               280.0
...               ...
20635           374.0
20636           150.0
20637           485.0
20638           409.0
20639           616.0

[20640 rows x 1 columns]

In [81]:
tbOut = df[df.isnull().any(axis=1)]

In [83]:
tbOut

longitude  latitude  ...  median_house_value  ocean_proximity
290      -122.16     37.77  ...            161900.0         NEAR BAY
341      -122.17     37.75  ...             85100.0         NEAR BAY
538      -122.28     37.78  ...            173400.0         NEAR BAY
563      -122.24     37.75  ...            247100.0         NEAR BAY
696      -122.10     37.69  ...            178400.0         NEAR BAY
...          ...       ...  ...                 ...              ...
20267    -119.19     34.20  ...            220500.0       NEAR OCEAN
20268    -119.18     34.19  ...            167400.0       NEAR OCEAN
20372    -118.88     34.17  ...            410700.0        <1H OCEAN
20460    -118.75     34.29  ...            258100.0        <1H OCEAN
20484    -118.72     34.28  ...            218600.0        <1H OCEAN

[207 rows x 10 columns]

In [84]:
len(tbOut.index)/len(df.index)*100

1.002906976744186

So total_bedrooms 207 rows are null values and less than 5%. Then we can't. reject these records. Then we have to impute.

In [92]:
df.groupby('total_bedrooms').head()

longitude  latitude  ...  median_house_value  ocean_proximity
0        -122.23     37.88  ...            452600.0         NEAR BAY
1        -122.22     37.86  ...            358500.0         NEAR BAY
2        -122.24     37.85  ...            352100.0         NEAR BAY
3        -122.25     37.85  ...            341300.0         NEAR BAY
4        -122.25     37.85  ...            342200.0         NEAR BAY
...          ...       ...  ...                 ...              ...
20543    -121.74     38.55  ...            182600.0           INLAND
20544    -121.76     38.55  ...            219400.0           INLAND
20563    -121.75     38.67  ...            132500.0           INLAND
20567    -121.79     38.66  ...            165000.0           INLAND
20629    -121.39     39.12  ...            108300.0           INLAND

[6652 rows x 10 columns]

1923 distinct values of total_bedrooms. So this is continuous value. The impute with mean / median.

In [95]:
df['total_bedrooms'].fillna(df['total_bedrooms'].mean(), inplace = True)

In [99]:
df[df.isnull().any(axis=1)]

Empty DataFrame
Columns: [longitude, latitude, housing_median_age, total_rooms, total_bedrooms, population, households, median_income, median_house_value, ocean_proximity]
Index: []

Now need to do normalisation.

longitude  latitude  ...  median_house_value  ocean_proximity
0    -122.23     37.88  ...            452600.0         NEAR BAY
1    -122.22     37.86  ...            358500.0         NEAR BAY
2    -122.24     37.85  ...            352100.0         NEAR BAY
3    -122.25     37.85  ...            341300.0         NEAR BAY
4    -122.25     37.85  ...            342200.0         NEAR BAY

[5 rows x 10 columns]